### PR TITLE
Use TaskCluster caching for Python and Homebrew setup on macOS

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

File: taskcluster/tc-brew-tests.sh [hunk 1]
```diff
@@ -24,15 +24,14 @@ install_local_homebrew()
 
     for suffix in .brew .cache .logs;
     do
-        if [ -d "${TASKCLUSTER_TASK_DIR}/${LOCAL_HOMEBREW_PREFIX}${suffix}/" ]; then
-            echo "Directory ${TASKCLUSTER_TASK_DIR}/${LOCAL_HOMEBREW_PREFIX}${suffix} already exists, aborting"
-            exit 1
+        if [ -d "${TASKCLUSTER_TASK_DIR}/homebrew.local/${LOCAL_HOMEBREW_PREFIX}${suffix}/" ]; then
+            echo "WARNING: Directory ${TASKCLUSTER_TASK_DIR}/homebrew.local/${LOCAL_HOMEBREW_PREFIX}${suffix} already exists"
```
Comment (reuben) [inline taskcluster/tc-brew-tests.sh]: Is this warning when we encounter the expected behavior? Why?

File: taskcluster/tc-tests-utils.sh [hunk 2]
```diff
@@ -714,7 +713,12 @@ pyenv_install()
       ln -s pip3.exe pip.exe
     popd
   else
-    pyenv install ${version}
+    # If there's already a matching directory, we should re-use it
+    # otherwise, pyenv install will force-rebuild
+    ls -hal "${PYENV_ROOT}/versions/${version}/" || true
```
Comment (reuben) [inline taskcluster/tc-tests-utils.sh]: Debugging leftover? Or is it intended?

Comment (lissyx) [reply taskcluster/tc-tests-utils.sh]: Debugging intended for now

File: taskcluster/tc-brew-tests.sh (hunk 1 above)
Comment (lissyx) [reply taskcluster/tc-brew-tests.sh]: That was a way to ensure finding that easily in the logs

**Pull request merged**